src/Application/Messenger/FundTotalUpdated.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace MatchBot\Application\Messenger;
+
+use MatchBot\Application\Assertion;
+use MatchBot\Domain\Fund;
+use Symfony\Component\Messenger\Bridge\AmazonSqs\MessageGroupAwareInterface;
+
+readonly class FundTotalUpdated implements MessageGroupAwareInterface
+{
+    public array $jsonSnapshot;
+
+    protected function __construct(public string $salesforceId, array $json)
```

Comment:
Not a but, might be worth documenting a more specific type than `array`. Less critical since I think this is only used in one or two places.

Also not sure if there's any reason to use the promoted property for the salesforceId but not for the jsonSnapshot

Reply:
True, I think it was a general `array` mostly because I first planned to extend `AbstractStateChanged` but that currently wants everything to have a UUID, and that didn't seem a good enough reason to give Funds an extra identifier.

I think because it's used quite narrowly I will probably leave it for now in the hopes of getting some version working today.

Good catch on the inconsistent promoted property use, I don't think there was a reason here or in `AbstractStateChanged`. Fixed both in https://github.com/thebiggive/matchbot/pull/1161/commits/355e05b884cdbea1f85f3dc38364e334a93c5b1b. This also helped me spot that there was a needless constructor in `DonationUpserted` so deleted that too.